PetrolEngine.cpp

```diff
@@ -15,6 +16,22 @@ void PetrolEngine::changeGear(int gear)
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+   
+    try
+    {
+        if (gear >= -1 && gear < gears_)
+        {
+            currentGear_ = gear;
+            std::cout << __FUNCTION__ << std::endl;
+        }
+        else throw std::out_of_range ("Invalid Gear");
+    }
+
+    catch (std::out_of_range  &text) { std::cout<<text.what()<<std::endl;}
+}
```

Comment:
np mam takie wymaganie do zaimplementowania:
"jesli gear sie nie ustawil to uzyj innego"
Jak takie cos moge zrobic uzywajac metody "changeGear"?

Reply:
Czyli chodzi o to, że łapanie wyjątku powinno być już w innym miejscu, tam gdzie zostanie wywołana funkcja `changeGear()`